radix-engine/src/blueprints/consensus_manager/consensus_manager.rs

```diff
@@ -532,19 +533,30 @@ impl ConsensusManagerBlueprint {
             api,
         )?;
 
-        // select next validator set
+        // Select next validator set
+        // NOTE - because the stake index is by u16 buckets, it's possible that there are multiple validators at the cut off point
+        // that fall into the same bucket, and it's possibly we arbitrarily (and incorrectly) choose a validator with lower stake,
+        // because they happen to have a higher DbSortKey in the index.
+        // This might cause us not to exactly select the top 100 validators. We accept this behaviour as a reasonable trade-off.
```

Comment:
wuuuuut only now I realized that the current impl in the Engine does not apply the solution that I discussed with Josh:
<img width="378" alt="image" src="https://github.com/radixdlt/radixdlt-scrypto/assets/119413677/d477950f-ad1b-4e04-89e9-cb4ca9a47d71">
 
:(

I looked at the call stack down the "list sorted" op, and I acknowledge it's not easy to return an iterator from there (on which the business logic here could "iterate until the stake bin changes again after 100"). But it seems possible, if we decided it's worth pursuing.

(I also acknowledge that this solution has a potential of reading unbounded number of entries from DB; like e.g. all of them, if all validators have close-enough stakes)

Do you think "it's worth it"?
- if yes: please transform this comment into a `TODO` and I can give it a try.
- if no: please surface the information from this comment up, to our public docs (I totally expect collisions within the "stake bins" we have, and being chosen into a validator set [or not] is probably serious enough)